src/aws_encryption_sdk/internal/formatting/encryption_context.py

```diff
@@ -47,9 +47,15 @@ def assemble_content_aad(message_id, aad_content_string, seq_num, length):
     return struct.pack(fmt, message_id, aad_content_string.value, seq_num, length)
 
 
-# help function to simplify code
-# this is necessary to pass flake8
+def _key_or_value_too_large(encryption_context_list):
+    """Helper method to check whether a key or value are too long."""
+    longest_key = max(encryption_context_list, key=lambda x: len(x[0]))[0]
```

Comment:
You are re-walking the list of keys and values to make this computation, but the list is already being walked at line 59. This is invoked at line 77 in an `except` block, which means you already are on a key or a value that is too large. Rather than re-walking the entire list, why not either 1) validate each item in line with the iteration on 59 or 2) check which of the current value(s) are too large in the `except`?

If you do want to walk the whole list, wouldn't it be better to do that right away to avoid wasting any work in the loop on 59?

Reply:
Added the check before packing.  Alternatively I could put this in the first run-through since there are two loops to fail faster.  The other part of the code deals with other exceptions, however, and I think it should be ok to leave it in the second loop for clarity.